CustomerService.UnitTests/Service/CustomerServiceTests.cs

```diff
@@ -0,0 +1,174 @@
+        TestCustomerEventConsumer consumer;
+
+        [TestInitialize]
+        public void Setup()
+        {
+            var host = new HostBuilder()
+         .ConfigureServices((hostContext, services) =>
+         {
+             //DbContext
+             services.AddDbContext<CustomerContext>(o => o.UseSqlServer(TestSettings.EventuateDB));
+             // Kafka Transport
+             services.AddEventuateTramSqlKafkaTransport(TestSettings.EventuateTramDbSchema, TestSettings.KafkaBootstrapServers, EventuateKafkaConsumerConfigurationProperties.Empty(),
+                (provider, o) =>
+                {
+                    var applicationDbContext = provider.GetRequiredService<CustomerContext>();
+                    applicationDbContext.Database.Migrate();
+                    o.UseSqlServer(applicationDbContext.Database.GetDbConnection());
+                });
+             // Publisher
+             services.AddEventuateTramEventsPublisher();
+             // Consumer
+             services.AddSingleton<TestCustomerEventConsumer>();
+             // Dispatcher
+             services.AddEventuateTramDomainEventDispatcher(Guid.NewGuid().ToString(),
+             provider => DomainEventHandlersBuilder.ForAggregateType(typeof(Customer).Name)
+              .OnEvent<CustomerCreatedEvent, TestCustomerEventConsumer>()
+              .OnEvent<CustomerValidationFailedEvent, TestCustomerEventConsumer>()
+              .OnEvent<CustomerCreditReservedEvent, TestCustomerEventConsumer>()
+              .OnEvent<CustomerCreditReservationFailedEvent, TestCustomerEventConsumer>()
+              .Build());
+             // Repository
+             services.AddTransient<ICustomerRepository, CustomerRepository>();
+         }).Build();
+            host.StartAsync().Wait();
+
+            //Services
+            domainEventPublisher = host.Services.GetService<IDomainEventPublisher>();
+            consumer = host.Services.GetService<TestCustomerEventConsumer>();
+            var customerRepository = host.Services.GetService<ICustomerRepository>();
+            customerDataService = new CustomerDataService(customerRepository, domainEventPublisher);
+            //Initialize Money
+            creditLimit = new Money("12.10");
+        }
+        [TestMethod]
+        public void CustomerShouldbeCreatedAndEventShouldbePublished()
+        {
+            //Create Customer
+            var customer = customerDataService.CreateCustomer("Joe", creditLimit);
+            // assert
+            Assert.IsNotNull(customer.Id);
+            CustomerCreatedEvent customerCreatedEvent;
+            consumer.GetCustomerCreatedEventQueue().TryTake(out customerCreatedEvent, TimeSpan.FromSeconds(20));
+
+            Assert.IsNotNull(customerCreatedEvent);
+            Assert.AreEqual("Joe", customerCreatedEvent.Name);
+        }
+
+        [TestMethod]
+        public void ReserveCreditAndEventShouldBePublished()
+        {
+            //Create Customer
```

Comment:
I think you can improve readability, reduce duplicate and eliminate the need for comments like

```
// Do something
... Some code...
```

By extracting lines of code into a new method

```
DoSomething(...) {
... Some code...
}
```

Reply:
Done